### PR TITLE
Add `chainerx.sigmoid_cross_entropy`

### PR DESCRIPTION
**Pull request by aksub99**
Add `chainerx.sigmoid_cross_entropy`.

### CODE REVIEW

**Comment by aksub99** [comment]
@hvy Could you please have a look at the definition in `loss.cc` and let me know if I'm going in the right direction?

File: chainerx_cc/chainerx/routines/loss.cc [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+#include "chainerx/routines/creation.h"
+#include "chainerx/routines/explog.h"
+#include "chainerx/routines/logic.h"
+#include "chainerx/routines/misc.h"
+#include "chainerx/routines/reduction.h"
+#include "chainerx/scalar.h"
+
+namespace chainerx {
+
+Array SigmoidCrossEntropy(const Array& x1, const Array& x2, bool normalize) {
```
Comment (crcrpar) [inline chainerx_cc/chainerx/routines/loss.cc]: I'm concerned with the absence of `'reduce'` argument which determines whether to reduce the shape of the input.

ref: https://docs.chainer.org/en/stable/reference/generated/chainer.functions.sigmoid_cross_entropy.html

Comment (aksub99) [reply chainerx_cc/chainerx/routines/loss.cc]: Yes @crcrpar . I do plan to add the implementation for `reduce` in this PR  Sorry I had forgotten to add a comment on the PR. I have added a TODO above.

Comment (hvy) [reply chainerx_cc/chainerx/routines/loss.cc]: We thought that the reduction argument complicates the interface (e.g. in Chainer, different functions offer different reductions but it doesn't have to be that way) and choose not to introduce it when not needed in a discussion here https://github.com/chainer/chainer/pull/7063#issuecomment-496087996 for ChainerX.

Comment (niboshi) [reply chainerx_cc/chainerx/routines/loss.cc]: I think `normalize` can also be omitted from the function.

Comment (aksub99) [reply chainerx_cc/chainerx/routines/loss.cc]: If normalize is omitted, what should be the default behavior of the function? (Normalize=True or Normalize=False)

Comment (niboshi) [reply chainerx_cc/chainerx/routines/loss.cc]: You can simply omit the division, as well as `Sum` and `ignore_mask *`.
That is, you can just return this expression: `-(x1 * (x2 - (GreaterEqual(x2, ZerosLike(x2)))) - Log1p(Exp(-Fabs(x1))))`

Comment (aksub99) [reply chainerx_cc/chainerx/routines/loss.cc]: Then the behavior wouldn't be identical to either of `normalize=True` or `normalize=False`. So, tests cannot use `ChainerOpTest`. So, would `NumpyOpTest` be preferred?

**Comment by aksub99** [comment]
Modifying tests to accommodate for the removal of `normalize` is still pending. @niboshi Could you please have a look at my comment above regarding tests?

File: chainerx_cc/chainerx/routines/loss.cc (hunk 1 above)
Comment (niboshi) [reply chainerx_cc/chainerx/routines/loss.cc]: You could either use `F.sigmoid_cross_entropy` with `normalize=False` and multiply the result by `len(x)`, or use `op_utils.OpTest`.

**Comment by aksub99** [comment]
I have fixed most of the errors. I still get the error 
`AttributeError: 'NoneType' object has no attribute 'shape'` in the gradient check.
@niboshi Could you please let me know of any possible cause?

**Comment by aksub99** [comment]
@niboshi Were you able to have a look?

**Comment by aksub99** [comment]
@niboshi I have fixed the tests. Could you please review?

**Comment by niboshi** [comment]
Jenkins, test this please

**Comment by niboshi** [comment]
Sorry for being late. The code LGTM and can be merged as soon as test passes.

**Comment by niboshi** [comment]
Jenkins failure is unrelated.
Thank you so much!

**Pull request merged**